miner/worker.go

```diff
@@ -908,6 +909,132 @@ func (w *worker) commitTransactions(txs *types.TransactionsByPriceAndNonce, coin
 	return false
 }
 
+// todo: commitL1Messages
+
+func (w *worker) commitL1Messages(l1Messages []types.L1MessageTx, coinbase common.Address, interrupt *int32) int64 {
+	// Short circuit if current is nil
+	if w.current == nil {
+		return -1
+	}
+
+	gasLimit := w.current.header.GasLimit
+	if w.current.gasPool == nil {
+		w.current.gasPool = new(core.GasPool).AddGas(gasLimit)
+	}
+
+	var coalescedLogs []*types.Log
+	var cntL1Messages int64 = 0
+
+	for _, l1Message := range l1Messages {
+		// In the following three cases, we will interrupt the execution of the transaction.
+		// (1) new head block event arrival, the interrupt signal is 1
+		// (2) worker start or restart, the interrupt signal is 1
+		// (3) worker recreate the mining block with any newly arrived transactions, the interrupt signal is 2.
+		// For the first two cases, the semi-finished work will be discarded.
+		// For the third case, the semi-finished work will be submitted to the consensus engine.
+		if interrupt != nil && atomic.LoadInt32(interrupt) != commitInterruptNone {
+			// Notify resubmit loop to increase resubmitting interval due to too frequent commits.
+			if atomic.LoadInt32(interrupt) == commitInterruptResubmit {
+				ratio := float64(gasLimit-w.current.gasPool.Gas()) / float64(gasLimit)
+				if ratio < 0.1 {
+					ratio = 0.1
+				}
+				w.resubmitAdjustCh <- &intervalAdjust{
+					ratio: ratio,
+					inc:   true,
+				}
+			}
+			if atomic.LoadInt32(interrupt) == commitInterruptNewHead {
+				return -1
+			} else {
+				return cntL1Messages
+			}
+		}
+		// todo: exclude l1Message txs from limit
+		// If we have collected enough transactions then we're done
+		if !w.chainConfig.IsValidTxCount(w.current.tcount + 1) {
+			log.Trace("Transaction count limit reached", "have", w.current.tcount, "want", w.chainConfig.MaxTxPerBlock)
+			break
+		}
+		// If we don't have enough gas for any further transactions then we're done
+		if w.current.gasPool.Gas() < params.TxGas {
+			log.Trace("Not enough gas for further transactions", "have", w.current.gasPool, "want", params.TxGas)
+			break
+		}
+		// Retrieve the next transaction and abort if all done
+		tx := types.NewTx(&l1Message)
+
+		// Start executing the transaction
+		w.current.state.Prepare(tx.Hash(), w.current.tcount)
+
+		logs, _ := w.commitTransaction(tx, coinbase)
+		cntL1Messages++
+		coalescedLogs = append(coalescedLogs, logs...)
+		w.current.tcount++
+		/*switch {
```

Comment:
IIUC from [this](
https://www.notion.so/scrollzkp/Bridge-Design-Doc-Before-4844-WIP-4f5a3130a94d49b9895e7237cdc0244f?pvs=4#7effe497963d452aaa0c8d4597734ff3) error handling is not needed here